src/main/docs/guide/kubernetes-informer.adoc

```diff
@@ -0,0 +1,63 @@
+|`resourcePlural`| The resource plural that identifies the Api. For example for the resource `V1ConfigMap` it is `configmaps`.
+|`namespace` | Namespace of the watched resource. If left empty then namespace is resolved by link:{api}/io/micronaut/kubernetes/client/NamespaceResolver.html[NamespaceResolver]. To watch resources from all namespaces configure this parameter to `Informer.ALL_NAMESPACES`.
+|`namespaces` | List of the namespaces of the watcher resource. The https://javadoc.io/doc/io.kubernetes/client-java/latest/io/kubernetes/client/informer/SharedIndexInformer.html[SharedIndexInformer] will be created for every namespace and all events will be delivered to the specified `ResourceEventHandler`.
+|`namespacesSupplier` | `Supplier` class that provides the list of namespaces to watch. Note that the supplier class needs to be a bean in the application context and it is intended for dynamic evaluation of the namesapces to watch. Finally the `namespace`, `namespaces` and `namespacesSupplier` can be used in combination.
+|`labelSelector` | Informer label selector, see https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors[Label selectors] for more information. By default there is no label selector.
+|`labelSelectorSupplier` | `Supplier` class for the label selector. Note that the supplier class needs to be a bean in the application context. Finally the `labelSelector` and `labelSelectorSupplier` can be used in combination.
+|`resyncCheckPeriod` | How often to check the need for resync of resources. If left empty the default resync check period is used.
+
+|===
+
+IMPORTANT: The concept of shared informer means that the https://javadoc.io/doc/io.kubernetes/client-java/latest/io/kubernetes/client/informer/SharedIndexInformer.html[SharedIndexInformer] for the respective Kubernetes resource type is registered just once for the given namespace. The next request to register another https://javadoc.io/doc/io.kubernetes/client-java/latest/io/kubernetes/client/informer/SharedIndexInformer.html[informer] of the same Kubernetes resource type within the same namespace will result in returning of the previously created informer. In practice if you create two `ResourceEventHandler<V1ConfigMap>` but the `@Informer` annotation will have different optional configuration for `labelSelector` then the `SharedInformerFactory` creates just one `SharedInformer`, meaning the other `@Informer` configuration will be ignored. If the `labelSelector` resp. `labelSelectorSupplier` differs then create one `labelSelector` that matches both cases.
+
+== Programmatic creation of SharedIndexInformer
+
+Use the bean link:{api}/io/micronaut/kubernetes/client/informer/SharedIndexInformerFactory[SharedIndexInformerFactory] to create the https://javadoc.io/doc/io.kubernetes/client-java/latest/io/kubernetes/client/informer/SharedIndexInformer.html[SharedIndexInformer] programmatically:
```

Comment:
note links can also be simplified with `api:kubernetes.client.informer.SharedIndexInformerFactory[]`